File: atlhyper_master_v2/model/cronjob.go
// atlhyper_master_v2/model/cronjob.go
// CronJob Web API 响应类型（camelCase JSON tag，扁平结构）
package model

// CronJobItem CronJob 列表项
type CronJobItem struct {
	Name               string `json:"name"`
	Namespace          string `json:"namespace"`
	Schedule           string `json:"schedule"`
	Suspend            bool   `json:"suspend"`
	ActiveJobs         int32  `json:"activeJobs"`
	LastScheduleTime   string `json:"lastScheduleTime"`
	LastSuccessfulTime string `json:"lastSuccessfulTime"`
	CreatedAt          string `json:"createdAt"`
	Age                string `json:"age"`
}

// CronJobDetail CronJob 详情
type CronJobDetail struct {
	// 基本信息
	Name      string `json:"name"`
	Namespace string `json:"namespace"`
	UID       string `json:"uid"`
	OwnerKind string `json:"ownerKind,omitempty"`
	OwnerName string `json:"ownerName,omitempty"`
	CreatedAt string `json:"createdAt"`
	Age       string `json:"age"`

	// 调度配置
	Schedule   string `json:"schedule"`
	Suspend    bool   `json:"suspend"`
	ActiveJobs int32  `json:"activeJobs"`

	// 时间
	LastScheduleTime   string `json:"lastScheduleTime"`
	LastSuccessfulTime string `json:"lastSuccessfulTime"`
	LastScheduleAgo    string `json:"lastScheduleAgo"`
	LastSuccessAgo     string `json:"lastSuccessAgo"`

	// 元数据
	Labels map[string]string `json:"labels,omitempty"`
}

File: atlhyper_web/src/api/cronjob.ts
/**
 * CronJob API
 *
 * 后端已完成 model_v2 → model 扁平化转换，前端直接使用
 */

import { get } from "./request";
import type { JobPodTemplate } from "./job";

// ============================================================
// 类型定义（匹配后端 model 响应）
// ============================================================

export interface CronJobItem {
  name: string;
  namespace: string;
  schedule: string;
  suspend: boolean;
  activeJobs: number;
  lastScheduleTime: string;
  lastSuccessfulTime: string;
  createdAt: string;
  age: string;
}

export interface CronJobDetail {
  name: string;
  namespace: string;
  uid: string;
  ownerKind?: string;
  ownerName?: string;
  createdAt: string;
  age: string;
  schedule: string;
  suspend: boolean;
  concurrencyPolicy?: string;
  activeJobs: number;
  successfulJobsHistoryLimit?: number;
  failedJobsHistoryLimit?: number;
  lastScheduleTime: string;
  lastSuccessfulTime: string;
  lastScheduleAgo: string;
  lastSuccessAgo: string;
  template?: JobPodTemplate;
  labels?: Record<string, string>;
  annotations?: Record<string, string>;
}

// ============================================================
// 响应类型（内部使用）
// ============================================================

interface ListResponse<T> {
  message: string;
  data: T[];
  total: number;
}

interface DetailResponse<T> {
  message: string;
  data: T;
}

// ============================================================
// API 查询参数（内部使用）
// ============================================================

interface ClusterResourceParams {
  cluster_id: string;
  namespace?: string;
}

// ============================================================
// API Functions
// ============================================================

export function getCronJobList(params: ClusterResourceParams) {
  return get<ListResponse<CronJobItem>>("/api/v2/cronjobs", params);
}

export function getCronJobDetail(params: { ClusterID: string; Namespace: string; Name: string }) {
  return get<DetailResponse<CronJobDetail>>(
    `/api/v2/cronjobs/${encodeURIComponent(params.Name)}`,
    { cluster_id: params.ClusterID, namespace: params.Namespace }
  );
}
